h2o-core/src/main/java/water/jdbc/SQLManager.java

```diff
@@ -46,7 +46,7 @@
       final Boolean useTempTable, final String tempTableName,
       final SqlFetchMode fetchMode, final Integer numChunksHint) {
 
-    final Key<Frame> destination_key = Key.make((table + "_sql_to_hex").replaceAll("\\W", "_"));
+    final Key<Frame> destination_key = Key.makeRandom(table, "sql_to_hex");
```

Comment:
Instead of using a partially randomized key, I think it is cleaner to follow a similar pattern as we use for naming models - just use an increasing sequence of numbers for a suffix, see H2O#calcNextUniqueModelId for inspiration. 

it could be even very simple, something like

    private static AtomicLong nextTableNum = new AtomicLong(0);
    ...
    final Key<Frame> destination_key = Key.make((table.replaceAll("\\W", "_") + "_sql_to_hex_" + nextTableNum.addAndGet(1)));

The advantage definitely is that it is clear in what order the keys were created - which could be useful in debugging.

Reply:
OK, you can have a look.